crates/wg_tests/src/test_fragments.rs

```diff
@@ -86,25 +92,29 @@ pub fn generic_fragment_drop<T: Drone + Send + 'static>() {
     // "Client" sends packet to the drone
     d_send.send(msg.clone()).unwrap();
 
-    let dropped = Nack {
-        fragment_index: 1,
-        nack_type: NackType::Dropped,
-    };
-    let srh = SourceRoutingHeader {
-        hop_index: 1,
-        hops: vec![11, 1],
-    };
-    let nack_packet = Packet {
-        pack_type: PacketType::Nack(dropped),
-        routing_header: srh,
-        session_id: 1,
-    };
+    let nack_packet = Packet::new_nack(
+        SourceRoutingHeader {
+            hop_index: 1,
+            hops: vec![11, 1],
+        },
+        1,
+        Nack {
+            fragment_index: 1,
+            nack_type: NackType::Dropped,
+        },
+    );
 
     // Client listens for packet from the drone (Dropped Nack)
     assert_eq!(c_recv.recv().unwrap(), nack_packet);
+    // SC listen for event from the drone
+    assert_eq!(
+        d_event_recv.recv().unwrap(),
+        DroneEvent::PacketDropped(nack_packet)
```

Comment:
It should be the packet dropped not the nack. We should also decide when sending the nack we should also send a drone event of packetSend